torch/cuda/nvtx.py

```diff
@@ -1,43 +1,18 @@
-import os
-import glob
-import ctypes
-import platform
+try:
+    from torch._C import _nvtx
+except ImportError:
+    class _NVTXStub(object):
+        @staticmethod
+        def _fail(*args, **kwargs):
+            raise RuntimeError("NVTX functions not installed. Are you sure you have a CUDA build?")
 
-lib = None
-
-__all__ = ['range_push', 'range_pop', 'mark']
-
-
-def windows_nvToolsExt_lib():
-    lib_path = windows_nvToolsExt_path()
-    if len(lib_path) > 0:
-        lib_name = os.path.basename(lib_path)
-        lib = os.path.splitext(lib_name)[0]
-        return ctypes.cdll.LoadLibrary(lib)
-    else:
-        return None
+        rangePushA = _fail
+        rangePop = _fail
+        markA = _fail
 
+    _nvtx = _NVTXStub()
 
-def windows_nvToolsExt_path():
-    WINDOWS_HOME = 'C:/Program Files/NVIDIA Corporation/NvToolsExt'
-    NVTOOLEXT_HOME = os.getenv('NVTOOLSEXT_PATH', WINDOWS_HOME)
-    if os.path.exists(NVTOOLEXT_HOME):
-        lib_paths = glob.glob(NVTOOLEXT_HOME + '/bin/x64/nvToolsExt*.dll')
```

Comment:
Same story for nvtoolsext. What's worse is that CUDA installation scripts don't put this one in `PATH`.